src/main/java/org/folio/service/processing/reader/MarcJsonReader.java

```diff
@@ -0,0 +1,68 @@
+package org.folio.service.processing.reader;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+
+public class MarcJsonReader implements SourceReader {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MarcJsonReader.class);
+  private JsonReader reader;
+  private int chunkSize;
+
+  public MarcJsonReader(File file, int chunkSize) {
+    this.chunkSize = chunkSize;
+    try {
+      this.reader = new JsonReader(new InputStreamReader(FileUtils.openInputStream(file)));
+    } catch (IOException e) {
+      LOGGER.error("Cannot initialize reader", e);
+      throw new RecordsReaderException(e);
+    }
+  }
+
+  @Override
+  public List<String> next() {
+    RecordsBuffer recordsBuffer = new RecordsBuffer(this.chunkSize);
+    try {
+      Gson gson = new GsonBuilder().create();
+      if (reader.peek().equals(JsonToken.BEGIN_ARRAY)) {
+        reader.beginArray();
+      }
```

Comment:
I expected to see an 'else' section here (with preparation reader to work with single json object)

Reply:
no need, JsonReader recognizes single json object by itself